src/interfaces/permissions/handlers/permissions-request.ts

```diff
@@ -5,12 +5,8 @@ import { canonicalAuth } from '../../../core/auth.js';
 import { MessageReply } from '../../../core/message-reply.js';
 import { PermissionsRequest } from '../messages/permissions-request.js';
 
-export const handlePermissionsRequest: MethodHandler = async (
-  tenant,
-  message,
-  messageStore,
-  didResolver
-): Promise<MessageReply> => {
+export const handlePermissionsRequest: MethodHandler = async (input): Promise<MessageReply> => {
+  const { tenant, message, messageStore, didResolver } = input;
```

Comment:
Because not all props are used in this handler. So letting the handler choose what it cares about through destructuring.

Reply:
im not suggesting that you destructure _all_ of the properties of `input`.  im suggesting that you do the destructure instead of even having an `input` parameter
```
export const handlePermissionsRequest: MethodHandler = async ({ tenant, message, messageStore, didResolver }): Promise<MessageReply> => {
```
that would make it _much_ easier for someone reading the code to immediately understand what (sub)parameters this function cares about instead of having to actually read any of the body (which is especially important for those that use code folding)

Reply:
Done!